apollo-ios-codegen/Sources/IR/IR+RootFieldBuilder.swift

```diff
@@ -136,19 +136,14 @@ class RootFieldBuilder {
   ) {
     addSelections(from: selectionSet, to: target, atTypePath: typeInfo)
 
-    self.hasDeferredFragments = {
-      switch typeInfo.scopePath.last.value.IsDeferred {
-      case .value(false):
-        return false
-      case .value(true), .if(_):
-        return true
-      }
-    }()
+    self.containsDeferredFragment = typeInfo.scope.scopePath.containsDeferredFragment
 
-    typeInfo.entity.selectionTree.mergeIn(
-      selections: target.readOnlyView,
-      with: typeInfo
-    )
+    if typeInfo.scope.scopePath.last.value.deferCondition == nil {
```

Comment:
This is the check to disable all merging of deferred selections. It will be refactored in the work Anthony is doing.